src/rust/plugin-work-queue/src/server.rs

```diff
@@ -24,111 +31,266 @@ use rust_proto::plugin_work_queue::{
+        Ok(Response::new(response.into()))
     }
 
     async fn get_execute_analyzer(
         &self,
-        _request: Request<GetExecuteAnalyzerRequestProto>,
+        request: Request<GetExecuteAnalyzerRequestProto>,
     ) -> Result<Response<GetExecuteAnalyzerResponseProto>, Status> {
-        todo!()
+        let request = request.into_inner();
+        let request: GetExecuteAnalyzerRequest =
+            request.try_into().map_err(PluginWorkQueueError::from)?;
+        let response = self.get_execute_analyzer(request).await?;
+        Ok(Response::new(response.into()))
     }
 
-    async fn acknowledge(
+    async fn acknowledge_generator(
         &self,
-        _request: Request<AcknowledgeRequestProto>,
-    ) -> Result<Response<AcknowledgeResponseProto>, Status> {
-        todo!()
+        request: Request<AcknowledgeGeneratorRequestProto>,
+    ) -> Result<Response<AcknowledgeGeneratorResponseProto>, Status> {
+        let request = request.into_inner();
+        let request: AcknowledgeGeneratorRequest =
+            request.try_into().map_err(PluginWorkQueueError::from)?;
+        let response = self.acknowledge_generator(request).await?;
+        Ok(Response::new(response.into()))
+    }
+
+    async fn acknowledge_analyzer(
+        &self,
+        request: Request<AcknowledgeAnalyzerRequestProto>,
+    ) -> Result<Response<AcknowledgeAnalyzerResponseProto>, Status> {
+        let request = request.into_inner();
+        let request: AcknowledgeAnalyzerRequest =
+            request.try_into().map_err(PluginWorkQueueError::from)?;
+        let response = self.acknowledge_analyzer(request).await?;
+        Ok(Response::new(response.into()))
     }
 }
 
-pub async fn exec_service() -> Result<(), Box<dyn std::error::Error>> {
+pub async fn exec_service(
+    service_config: PluginWorkQueueServiceConfig,
+) -> Result<(), Box<dyn std::error::Error>> {
     let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
     health_reporter
         .set_serving::<PluginWorkQueueServiceServer<PluginWorkQueue>>()
         .await;
 
-    let addr = "[::1]:50051".parse().unwrap();
-    let plugin_work_queue = PluginWorkQueue {};
-
     tracing::info!(
-        message="HealthServer + PluginWorkQueue listening",
-        addr=?addr,
+        message="Connecting to plugin registry table",
+        service_config=?service_config,
     );
+    let postgres_address = format!(
+        "postgresql://{}:{}@{}:{}",
+        service_config.plugin_work_queue_db_username,
+        service_config.plugin_work_queue_db_password,
+        service_config.plugin_work_queue_db_hostname,
+        service_config.plugin_work_queue_db_port,
+    );
+
+    let plugin_work_queue: PluginWorkQueue = PluginWorkQueue::from(
+        sqlx::PgPool::connect(&postgres_address)
+            .timeout(std::time::Duration::from_secs(5))
+            .await??,
+    );
+
+    sqlx::migrate!().run(&plugin_work_queue.queue.pool).await?;
```

Comment:
So wait, every time this service starts it runs a migration? If so that seems wrong.

Reply:
It will first ask the db which migrations have already applied and not run any that have already been applied. This is apparently standard practice.